src/graphics/gui.c

```diff
@@ -12,19 +15,64 @@
 int screen_width = INITIAL_SCREEN_WIDTH;
 int screen_height = INITIAL_SCREEN_HEIGHT;
 char title[] = GAME_TITLE;
-
 int state = STATE_START;
 
-void runGameGui(){
+void runGameGui() {
+
     MenusBundle *menu_bundle = initMenusBundle(&screen_width, &screen_height);
+    GameplayData *gameplay_data = NULL;
+
     InitWindow(screen_width, screen_height, title);
+    Vector2 mouse_position = {0.0f, 0.0f};
     SetTargetFPS(60);
-    Vector2 mousePos = {0.0f, 0.0f};
-    Vector2 *mp = &mousePos;
-    while (!WindowShouldClose() && state != STATE_EXIT){
-        handleStateClicks(menu_bundle, mp, &state);
-        handleGUIDraw(menu_bundle, mp, &state, &screen_width, &screen_height);
+
+    // TODO: Move to a different file.
+    AssetManager *assets = (AssetManager*) malloc(sizeof(AssetManager));
+    assets->tile_1 = LoadTexture("assets/tile-1.png");
+    assets->tile_2 = LoadTexture("assets/tile-2.png");
+    assets->tile_3 = LoadTexture("assets/tile-3.png");
+    assets->tile_4 = LoadTexture("assets/tile-4.png");
+    assets->tile_5 = LoadTexture("assets/tile-5.png");
+    assets->tile_opened = LoadTexture("assets/tile-opened.png");
+    assets->tile_closed = LoadTexture("assets/tile-closed.png");
+    assets->tile_flag = LoadTexture("assets/tile-flag.png");
+    assets->tile_bomb = LoadTexture("assets/tile-bomb.png");
+
+    while (!WindowShouldClose() && state != STATE_EXIT) {
+
+        handleStateClicks(menu_bundle, &mouse_position, &state);
+
+        // Pre-drawing operations based on current state.
+        switch (state) {
```

Comment:
Would it be more clear if these switch statements were to be moved to separate functions? I think the purpose of Gui.c is to serve as a wrapper for the functions which are called based on the state.

Reply:
We want to minimize the number of checks performed at any given time... if we have a switch statement inside every function and call each of those functions, the number of checks is going to be very high. `gui.c`, as I see it, checks the current state and calls the function that has to be called. It performs the "routing".

Reply:
I know that there's a combination of ways that this is handled since the start menu does the function calling. We can discuss this in a meeting and create a ticket to clean up this code.